control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -257,59 +297,72 @@ impl SwitchOverEngine {
             let mut lo = recv.lock().await;
             if let Some(mut q) = lo.recv().await {
+                drop(lo);
                 info!(
-                    "switchover for vol: {} picked by worker-{}",
-                    q.volume_id, worker_num
+                    volume.uuid = %q.volume_id,
+                    worker = worker_num,
+                    "Volume switchover picked by worker",
                 );
-                drop(lo);
-                let retry_delay_min = std::time::Duration::from_secs(1);
-                let retry_delay_max = std::time::Duration::from_secs(384);
-                let mut retry_delay = retry_delay_min;
                 loop {
                     // TODO: handle error handling properly
                     let result = match q.stage {
                         Stage::Init => q.initialize(&self.etcd).await,
                         Stage::RepublishVolume => q.republish_volume(&self.etcd).await,
                         Stage::PublishPath => q.publish_path(&self.etcd, &self.nodes).await,
                         Stage::DeleteTarget => q.delete_target(&self.etcd).await,
+                        Stage::Errored => match q.errored_switchover(&self.etcd).await {
+                            Ok(_) => break,
+                            Err(e) => Err(e),
+                        },
                         Stage::Successful => match q.delete_switchover(&self.etcd).await {
                             Ok(_) => break,
                             Err(e) => Err(e),
                         },
-                        _ => break,
                     };
                     match result {
                         Ok(_) => {
-                            // reset retry delay back to the start.
-                            retry_delay = retry_delay_min;
+                            // reset retry count back to the start after successfully completing a
+                            // stage.
+                            q.retry_count = 0;
                         }
                         Err(_) => {
-                            retry_delay = (retry_delay * 3).min(retry_delay_max);
-                            tokio::time::sleep(retry_delay).await;
+                            info!(volume.uuid=%q.volume_id, "Sending failed Switchover request back to the channel");
+                            q.retry_count += 1;
+                            let _ = self.enqueue(q);
+                            break;
                         }
                     }
                 }
             }
         }
     }
 
-    pub fn initiate(&self, req: SwitchOverRequest) -> anyhow::Result<()> {
-        self.channel
-            .send(req)
-            .map_err(|error| anyhow::anyhow!("Failed to send switchover request: {error}"))
+    /// Sends Switchover request to the Channel after sleeping for a specified duration if
+    /// specified.
+    pub fn enqueue(&self, req: SwitchOverRequest) -> anyhow::Result<()> {
```

Comment:
this doesn't return any errors? So doesn't need to return anything?

Reply:
Removed returns